src/main/java/graphql/execution/incremental/IncrementalCallState.java

```diff
@@ -85,13 +87,19 @@ public boolean getIncrementalCallsDetected() {
         return incrementalCallsDetected.get();
     }
 
+    private Supplier<SingleSubscriberPublisher<DelayedIncrementalPartialResult>> createPublisher() {
+        // this will be created once and once only - any extra calls to .get() will return the previously created
+        // singleton object
+        return new InterThreadMemoizedSupplier<>(() -> new SingleSubscriberPublisher<>(this::drainIncrementalCalls));
```

Comment:
now this creates a publisher that only calls `drainIncrementalCalls` when the publisher is subscribed to